packages/autocomplete-core/src/__tests__/getAutocompleteSetters.test.ts

```diff
@@ -57,22 +80,56 @@ describe('createAutocomplete', () => {
     );
   });
 
-  test('setSuggestions', () => {
-    const onStateChange = jest.fn();
-    const { setSuggestions } = createAutocomplete({
-      getSources: () => [],
-      onStateChange,
+  describe('setSuggestions', () => {
+    test('default', () => {
+      const onStateChange = jest.fn();
+      const { setSuggestions } = createAutocomplete({
+        getSources: () => [],
+        onStateChange,
+      });
+
+      setSuggestions([createSuggestion([{ label: 'hi' }])]);
+
+      expect(onStateChange).toHaveBeenCalledTimes(1);
+      expect(onStateChange).toHaveBeenCalledWith({
+        state: expect.objectContaining({
+          suggestions: [
+            {
+              items: [
+                {
+                  label: 'hi',
+                  __autocomplete_id: 0,
+                },
+              ],
+              source: expect.any(Object),
+            },
+          ],
+        }),
+      });
     });
-    const suggestions = [createSuggestion()];
 
-    setSuggestions(suggestions);
+    test('flattens suggestions', () => {
+      const onStateChange = jest.fn();
+      const { setSuggestions } = createAutocomplete({
+        openOnFocus: true,
+        getSources: () => [],
+        onStateChange,
+      });
 
-    expect(onStateChange).toHaveBeenCalledTimes(1);
-    expect(onStateChange).toHaveBeenCalledWith(
-      expect.objectContaining({
-        state: expect.objectContaining({ suggestions }),
-      })
-    );
+      // while it also accepts a nested array.
+      // There's only one type for both input and output however,
+      setSuggestions([createMultiSuggestion([[{ label: 'hi' }]])]);
```

Comment:
Why can't we use `createSuggestion` here?

Reply:
createSuggestion expected the input to be an array of the item type, but here it's a nested array, and I couldn't see how to make it work with the internals :)

Reply:
Can't we accept `TItem[] | Array<TItem[]>` in `createSuggestion`? (That would better reflect how you use `getSuggestions`.)

Reply:
Then we need to accept items[][] as part of AutocompleteSuggestion, which is normally the type for the internal suggestions (thus already flattened), and I can't find which type to use which accepts the right type. Could you try to find the exact correct type in this branch?

Reply:
Here's a solution that would work:

```ts
function createSuggestion<TItem extends { label: string }>(
  items: TItem[] | TItem[][] = []
): AutocompleteSuggestion<TItem | TItem[]> | AutocompleteSuggestion<TItem[]> {
  return {
    source: {
      getInputValue: ({ suggestion }) => suggestion.label,
      getSuggestionUrl: () => undefined,
      onHighlight: () => {},
      onSelect: () => {},
      getSuggestions: () => items,
    },
    items,
  };
}
```